x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/expression/predicate/RangeTests.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ql.expression.predicate;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.ql.expression.Literal;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.type.DataType;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+import org.elasticsearch.xpack.ql.type.DateUtils;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+
+public class RangeTests extends ESTestCase {
+
+    private static final DataType K = DataTypes.KEYWORD;
+    private static final DataType T = DataTypes.TEXT;
+    private static final DataType DT = DataTypes.DATETIME;
+
+    public void testAreBoundariesInvalid() {
+        // value, value type, lower, lower type, lower included, higher, higher type, higher included, boundaries invalid
+        Object[][] tests = {
+            // dates
+            { d("2021-01-01"), DT, "2021-01-01", K, randomBoolean(), "2022-01-01", K, randomBoolean(), false },
+            { d("2021-01-01"), DT, "2022-01-01", K, randomBoolean(), "2021-01-01", K, randomBoolean(), true },
+            { d("2021-01-01"), DT, "now-10y", K, randomBoolean(), "2022-01-01", K, randomBoolean(), false },
```

Comment:
It doesn't matter if the text field type is keyword or text, so why not use everywhere `randomFrom(TEXT, KEYWORD)`? Extract this in a method `randomTextType()` and use that everywhere.